cumulus/components/userdata/linux.py

```diff
@@ -4,35 +4,37 @@
 )
 
 
-def user_data_for_cfn_init(launch_config_name, asg_name, configsets):
-    """
-    :return: A troposphere Join object that contains userdata for use with cfn-init
+class LinuxUserData:
```

Comment:
I'm not 100% sure this will or won't work for AmazonLinux & AmazonLinux2.   They are both somewhat compatible with RedHat from the perspective.  Maybe we need tighter coupling between the OS type and the userdata script.. or maybe we can make this compatible for both. 

I kind of prefer the latter as CentOS and AmazonLinux are similar.   

No action required,  just babbling commentary.

Reply:
I concur.  It was developed with centos in mind, but we could create one for other flavours.